src/ripple/app/tx/impl/InvariantCheck.h

```diff
@@ -365,6 +365,32 @@ class NFTokenCountTracking
         beast::Journal const&);
 };
 
+/**
+ * @brief Invariant: Trust line balance cannot be negative after Clawback.
+ *
+ * We iterate all the trust lines affected by this transaction and ensure
+ * that they have non-negative balance.
+ */
+class ValidClawback
+{
+    std::uint32_t trustlinesChanged = 0;
```

Comment:
My understanding of the `ValidClawback::visitEntry` function is that, `trustlinesChanges` can only be `0` or `1` (because of the said function definition in implementation file of `InvariantCheck`). 

Am I missing something? Hypothetically, even if more than one trust line were modified, the `ValidClawback::visitEntry()` would be called only once right?

Reply:
From my understanding, I think `ValidClawback::visitEntry` will iterated on every ledger object that was changed as a result of the transaction. And `trustlinesChanged` is incremented each time it encounters a `RIPPLE_STATE`(trustline) object. 

For a `Clawback` transaction, `trustlinesChanged` could only be 0 or 1. This is why in `ValidClawback::finalize`, there is a guard to check for the transaction type:
```
    if (tx.getTxnType() != ttCLAWBACK)
        return true;
```

On the other hand, it's possible that other type of transactions(I think `Payment`?) could potentially change 2 or more trustline objects in the transaction. In that case, `trustlinesChanged` could be some number greater than 1 (since `visitEntry` is iterated on modified objects for _all_ transaction types). 

In your hypothetical scenario, if more than one trustlines are changed, `visitEntry` will keeps count of the total number of modified trustlines as it iterates through all the objects that were changed

```
    if (before && before->getType() == ltRIPPLE_STATE)
        trustlinesChanged++;
```